src/sentry/integrations/pagerduty/notify_action.py

```diff
@@ -62,10 +62,11 @@ def clean(self):
         return cleaned_data
 
 
-class PagerDutyNotifyServiceAction(EventAction):
+class PagerDutyNotifyServiceAction(IntegrationEventAction):
```

Comment:
@mgaeta did you test these changes with all the integrations that were modified?

Reply:
I wasn't able to install PagerDuty, Slack, or MS Teams locally to test. It looks like existing tests pass though.

Reply:
@mgaeta I can try to help you get those running if you'd like. If not, then we should just not make the changes for those integrations and put a TODO to tackle later. Don't want to risk a change for critical functionality that we aren't testing. I'm not 100% sure that existing tests would catch any potential issue here.

Reply:
I've pulled out the refactor commits into this PR: https://github.com/getsentry/sentry/pull/21646.